### ISSUE TITLE
Recherche

### ISSUE THREAD

**Issue by goetsu** [osunyorg/theme#748]
- à la fermeture de la recherche remettre le focus sur le bouton "rechercher" qui déclenche l'affichage de la modale de recherche (conformité)
- ajouter un `<p role="heading" aria-level="1" class="sr-only">Recherche</p>` après le bouton de fermeture de la modale (amélioration)
- quand l’utilisateur ferme la recherche vider la div `<div aria-live="polite" aria-atomic="true" class="sr-only pagefind-ui__accessible-message">` (conformité)
- dans les résultats recherche le titre de chaque bloc de résultat p.pagefind-ui__result-title devrait etre balisé avec un `<h2>` (conformité)
- gestion du alt des images dans les résultats recherche devrait etre vide alt="" (point à discuter pour arbitrage)

**Comment by arnaudlevy** [comment]
## Problème

### URL

Sur quelle URL peut-on observer le problème ?

TODO

### Description

Qu'est-ce qui ne va pas ? 

TODO

### Impact

Dans quelle situation ceci est-il gênant ? En vocal ? Au clavier ? Quel handicap ?

TODO

### Nature

Est-ce un problème de conformité RGAA ou une proposition d'amélioration ?
Si c'est un problème de conformité, quel est le critère concerné ?

TODO

### Code problématique

Code, éventuellement simplifié, à l'origine du problème.

TODO

```HTML
```

## Solution

### Description

En quoi consiste la solution ?

TODO

### Code cible

À quoi doit ressembler le code après réparation ?

TODO

```HTML
```

### Effet attendu

En quoi est-ce une amélioration pour les personnes impactées ?
Par exemple, quelle est la vocalisation attendue ?

TODO

### Source

D'où vient cette recommandation ? 

TODO

### Exemple d'implémentation

Où peut-on observer cette solution en production ?

TODO

**Comment by arnaudlevy** [comment]
https://github.com/CloudCannon/pagefind/issues/727

**Comment by alexisben** [comment]
| - ajouter un "<p role="heading" aria-level="1" class="sr-only">Recherche</p>" après le bouton de fermeture de la modale (amélioration)

Il me semble que cela ajouterai un second H1 dans l'arborescence des titres ce qui contrevient aux recommandations de la structuration pertinent : [https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#9.1.1](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#9.1.1)

**Comment by arnaudlevy** [comment]
Point 2 : cela créerait 2 h1, ce qui est déconseillé par [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#avoid_using_multiple_h1_elements_on_one_page)

**Comment by arnaudlevy** [comment]
Point 1 : @goetsu quel est le critère qui demande ça ?

**Comment by arnaudlevy** [comment]
Les points 4 et 5 dépendent de Pagefind

**Comment by alexisben** [comment]
@goetsu je ne sais pas si tu as vu nos commentaires ici ?

**Comment by goetsu** [comment]
point 1 critère 12.8
point 2 le contenu de la recherche est une modale l'arrière plan est donc non restitué il n'y aura donc jamais 2 h1 (il n'est par ailleurs en rien interdit par le RGAA d'avoir plusieurs h1 dans une page (ni par wcag https://www.tpgi.com/heading-off-confusion-when-do-headings-fail-wcag/)

**Comment by alexisben** [comment]
Merci pour le critère ! 

Pour le point 1 et 3 c'est corrigé.

Pour le point 2 et 4 je comprends tout à fait le point de vue en mode lecteur d'écran. Mais je suis embêté car ça met le titre principal de la page au même niveau que "Rechercher" qui est présent sur toutes les pages, sémantiquement ça semble problématique ? 

Pour le point 5 je ne comprends pas pourquoi il faut des `alt` vides ? Car c'est un contexte de résultat de recherche ?

**Comment by goetsu** [comment]
la modale est en display none dans le code donc encore une fois il n'y a pas deux h1 après si c'est pour des raisons SEO tu peux utiliser dans la modale `<p role="heading" aria-level="1">` à la place du h1

pour le point 5 peut tu me dire comment les images sont récupérés ? c'est un contribution spécifique ? c'est la première image dans l'article ?

**Comment by alexisben** [comment]
Ok je crois que je comprends, le `role`et l'`aria-level` ne sont considérés que par les technologies d'assistances, donc ça ne change pas la sémantique de la page web au sens du HTML sémantique pur ?

Pour le point 5 ce n'est pas une contribution spécifique à la recherche. C'est la même image que l'image de partage `og:image`, c'est à dire l'image principale de la page (celle dans le heros), ou l'image de partage par défaut du site (contribuée) s'il n'y pas d'image associée à la page.

**Comment by goetsu** [comment]
oui c'est ça pour le role et le aria-level

pour les images exemple sur https://example.osuny.org/fr/actualites/2023-03-04-referentiel-general-decoconception-de-services-numeriques-rgesn/ du fait que l'image ouvre dans la visionneuse on va recommander au contributeur qu'elle dispose d'un alt décrivant l'image car dans le contexte visionneuse c'est nécessaire cf discussion ticket block image sauf qu'ici dans le contexte résultat recherche cela n'a aucune intéret de décrire l'image elle n'ouvre pas la visionneuse. 

Après si sur certains article on a une image réellement informative utilisée en hero d'un article sans visionneuse, le alt dans le résultat de recherche pourrait etre utile

Je ne vois pas de solution réellement tout terrain mais vu l'usage ici en mode vignette réduite + le fait que le lien permet de toute façon d'accéder à l'article ou si besoin l'utilisateur va trouver l'image et son alt complet il est préférable de partir sur un alt="" en dur dans le template

**Comment by alexisben** [comment]
Ok je comprends mieux merci. 

Pour le `alt` je regarde ce que l'on peut faire sur pagefind (hack ou soumission).

**Comment by alexisben** [comment]
On est bon pour ces 5 points sur example !

**Comment by goetsu** [comment]
ok pour moi
